reorganized the first section, in line with MEG and EEG spec (#2)
choldgraf/bids-specification: src/04-modality-specific-files/06-intracranial-electroencephalography.md

of the academic literature:

> TODO: citation

## iEEG recording data

Template:

```Text
sub-<label>/
  [ses-<label>]/
    ieeg/
      sub-<label>[_ses-<label>]_task-<task_label>[_acq-<label>][_run-<index>]_ieeg.<manufacturer_specific_extension>
      sub-<label>[_ses-<label>]_task-<task_label>[_acq-<label>][_run-<index>]_ieeg.json
```

The iEEG community uses a variety of formats for storing raw data, and there is
no single standard that all researchers agree on. The allowed file formats for
iEEG data in BIDS are divided into two groups: _recommended_ formats that are
open, well-defined standards, and _accepted_ formats that are common in the
community. In general, it is discouraged to use _accepted_ formats over
_recommended_ formats, particularly because there are conversion scripts
available in most analytics languages to convert data into _recommended_
formats. Below are lists of each group of allowed data formats in BIDS-iEEG.

**Recommended Formats**

-   European Data Format (.edf) (https://www.edfplus.info/)
-   Brainvision (.vhdr/.eeg/.vmrk)
    [BrainVision data format](https://www.brainproducts.com/productdetails.php?id=21&tab=5)
    by Brain Products GmbH.

> **Allowed Formats**

Reviewer: ```suggestion
**Accepted Formats**
```

Reviewer: just for consistency's sake since we use this word above

-   [Neurodata Without Borders](<(https://github.com/NeurodataWithoutBorders/pynwb)>)
    (.nwb)
-   [EEGLAB](https://sccn.ucsd.edu/eeglab) (`.set` and `.fdt` files)
-   [MEF3](https://github.com/msel-source) (`.mef`)
    ([paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4956586/),
    [specification and links](http://msel.mayo.edu/codes.html),
    [Python reader](https://www.google.com/url?q=https://github.com/ICRC-BME/PySigView&sa=D&ust=1540234072960000&usg=AFQjCNEBg8ua_qEc7U3OsK0lh-iXn94mWQ))

Future versions of BIDS may extend this list of supported file formats. File
formats for future consideration MUST have open access documentation, MUST have
open source implementation for both reading and writing in at least two
programming languages and SHOULD be widely supported in multiple software
packages. Other formats that may be considered in the future should have a clear
added advantage over the existing formats and should have wide adoption in the
BIDS-iEEG community.

> The original data format is especially valuable in case conversion elicits the

Reviewer: ```suggestion
The data format in which the data was originally stored is especially valuable in case conversion elicits the
```

loss of crucial metadata specific to manufacturers and specific iEEG systems. We
also encourage users to provide additional meta information extracted from the

> manufacturer specific data files in the sidecar JSON file. Other relevant files

Reviewer: ```suggestion
manufacturer-specific data files in the sidecar JSON file. Other relevant files
```

> MAY be included alongside the original EEG data in the

Reviewer: I guess it's fine to say EEG since the intracranial is implied?

[`/sourcedata` directory](../02-common-principles.md#source-vs-raw-vs-derived-data).

> For proper documentation of iEEG recording metadata it is important to

Reviewer: ```suggestion

### Terminology: Electrodes vs. Channels

For proper documentation of iEEG recording metadata it is important to
```
What do you think about this, just to make the page easier to parse?

understand the difference between electrode and channel: An EEG electrode is
attached to the skin, whereas a channel is the combination of the analog

> differential amplifier and analog-to-digital converter that result in a

Reviewer: ```suggestion
differential amplifier and analog-to-digital converter that results in a
```

potential (voltage) difference that is stored in the EEG dataset. We employ the
following short definitions:

> -   Electrode = A single point of contact between the acquisition system and the

Reviewer: ```suggestion
-   **Electrode** = A single point of contact between the acquisition system and the
```

    recording site (e.g., scalp, neural tissue, ...). Multiple electrodes can be
    organized as arrays, grids, leads, strips, probes, shafts, caps (for EEG),
    etc.

> -   Channel = A single analog-to-digital converter in the recording system that

Reviewer: ```suggestion
-   **Channel** = A single analog-to-digital converter in the recording system that
```

    regularly samples the value of a transducer, which results in the signal
    being represented as a time series in the digitized data. This can be
    connected to two electrodes (to measure the potential difference between
    them), a magnetic field or magnetic gradient sensor, temperature sensor,
    accelerometer, etc.

Although the _reference_ and _ground_ electrodes are often referred to as
channels, they are in most common iEEG systems not recorded by themselves.
Therefore they are not represented as channels in the data. The type of
referencing for all channels and optionally the location of the reference
electrode and the location of the ground electrode MAY be specified.

> ### Sidecar JSON document (`*_ieeg.json`)

Reviewer: ```suggestion

### Sidecar JSON document (`*_ieeg.json`)
```

For consistency between studies and institutions, we encourage users to extract
the values of metadata fields from the actual raw data. Whenever possible,
please avoid using ad hoc wording.

> Generic fields MUST be present:

Reviewer: ```suggestion
**Generic fields**

MUST be present:
```

| SamplingFrequency  | REQUIRED. Sampling frequency (in Hz) of all the iEEG channels in the recording (e.g., 2400). All other channels should have frequency specified as well in the `channels.tsv` file.                                                                                                                                                                                                                                     |
| PowerLineFrequency | REQUIRED. Frequency (in Hz) of the power grid where the iEEG recording was done (i.e., 50 or 60).                                                                                                                                                                                                                                                                                                                       |

> | SoftwareFilters    | REQUIRED. List of temporal software filters applied or ideally key:value pairs of pre-applied filters and their parameter values. (n/a if none). E.g., “{'HighPass': {'HalfAmplitudeCutOffHz': 1, 'RollOff: '6dB/Octave'}}”.                                                                                                                                                                                            |

Reviewer: ```suggestion
| SoftwareFilters    | REQUIRED. List of temporal software filters applied or ideally key:value pairs of pre-applied filters and their parameter values. (n/a if none). E.g., `{'HighPass': {'HalfAmplitudeCutOffHz': 1, 'RollOff: '6dB/Octave'}}`.                                                                                                                                                                                            |
```

> SHOULD be present: For consistency between studies and institutions, we

Reviewer: ```suggestion
**SHOULD be present**

For consistency between studies and institutions, we
```

| CogPOID                | RECOMMENDED. URL of the corresponding [CogPO](http://www.cogpo.org/) term.                                                                                                                                                    |
| InstitutionName        | RECOMMENDED. The name of the institution in charge of the equipment that produced the composite instances.                                                                                                                    |
| InstitutionAddress     | RECOMMENDED. The address of the institution in charge of the equipment that produced the composite instances.                                                                                                                 |
| DeviceSerialNumber     | RECOMMENDED. The serial number of the equipment that produced the composite instances. A pseudonym can also be used to prevent the equipment from being identifiable, as long as each pseudonym is unique within the dataset. |

> Specific iEEG fields MUST be present:

Reviewer: ```suggestion
**Specific iEEG fields**

MUST be present:
```

| :------------ | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| iEEGReference | REQUIRED. General description of the reference scheme used and (when applicable) of location of the reference electrode in the raw recordings (e.g., "left mastoid”, “bipolar”, “T01” for electrode with name T01, “intracranial electrode on top of a grid, not included with data”, “upside down electrode”). If different channels have a different reference, this field should have a general description and the channel specific reference should be defined in the \_channels.tsv file. |

> Specific iEEG fields SHOULD be present:

Reviewer: ```suggestion
**SHOULD be present**
```

| iEEGGround                      | RECOMMENDED. Description of the location of the ground electrode (“placed on right mastoid (M2)”).                                                                                                                                                                                       |
| iEEGPlacementScheme             | RECOMMENDED. Freeform description of the placement of the iEEG electrodes. Left/right/bilateral/depth/surface (e.g., “left frontal grid and bilateral hippocampal depth” or “surface strip and STN depth” or “clinical indication bitemporal, bilateral temporal strips and left grid”). |
| iEEGElectrodeGroups             | RECOMMENDED. Field to describe the way electrodes are grouped into strips, grids or depth probes e.g., {'grid1': "10x8 grid on left temporal pole", 'strip2': "1x8 electrode strip on xxx"}.                                                                                             |

> Specific iEEG fields MAY be present:

Reviewer: ```suggestion
**MAY be present**
```

`YYYY-MM-DDThh:mm:ss`
([ISO8601](https://en.wikipedia.org/wiki/ISO_8601) date-time format). For
example: 2009-06-15T13:45:30. It does not need to be fully detailed, depending
on local REB/IRB ethics board policy.

## Channels description table (`*_channels.tsv`)

Template:

```Text
sub-<label>/
    [ses-<label>]/
      ieeg/
        [sub-<label>[_ses-<label>]_task-<label>[_acq-<label>][_run-<index>]_channels.tsv]
```

A channel represents one time series recorded with the recording system (for
example, there can be a bipolar channel, recorded from two electrodes or contact
points on the tissue). Although this information can often be extracted from the
iEEG recording, listing it in a simple `.tsv` document makes it easy to browse
or search (e.g., searching for recordings with a sampling frequency of >=1000
Hz). The two required columns are channel name and type. Channels should appear
in the table in the same order they do in the iEEG data file. Any number of
additional columns may be provided to provide additional information about the
channels. Note that electrode positions should not be added to this file but to
`*_electrodes.tsv`.

> The columns of the Channels description table stored in \*\_channels.tsv are:

Reviewer: ```suggestion
The columns of the Channels description table stored in `*_channels.tsv` are listed below.
```

> MUST be present:

Reviewer: ```suggestion
**MUST be present**
```

| units       | REQUIRED. Physical unit of the value represented in this channel, e.g., V for Volt, specified according to the [SI unit symbol](https://en.wikipedia.org/wiki/International_System_of_Units#Base_units) and possibly prefix symbol (e.g., mV, μV), see [TODO: bids link] the BIDS spec (section 15 Appendix V: Units) for guidelines for Units and Prefixes. |
| low_cutoff  | REQUIRED. Frequencies used for the low pass filter applied to the channel in Hz. If no low pass filter was applied, use `n/a`. Note that anti-alias is a low pass filter, specify its frequencies here if applicable.                                                                                                                                        |
| high_cutoff | REQUIRED. Frequencies used for the high pass filter applied to the channel in Hz. If no high pass filter applied, use `n/a`.                                                                                                                                                                                                                                 |

> SHOULD be present:

Reviewer: ```suggestion
**SHOULD be present**
```

| notch              | OPTIONAL. Frequencies used for the notch filter applied to the channel, in Hz. If no notch filter applied, use n/a.                                                                                                                                                                  |
| status             | OPTIONAL. Data quality observed on the channel (good/bad). A channel is considered bad if its data quality is compromised by excessive noise. Description of noise type SHOULD be provided in `[status_description]`.                                                                |
| status_description | OPTIONAL. Freeform text description of noise or artifact affecting data quality on the channel. It is meant to explain why the channel was declared bad in [status].                                                                                                                 |

**Example** `sub-01_channels.tsv`:

```Text
name  type  units sampling_frequency low_cutoff high_cutoff notch group reference description              status status_description

> LT01  ECOG  μV    10000             3000       0.11        n/a   LTG   mastoid   lateral_temporal_surface good   n/a

Reviewer: ```suggestion
LT01  ECOG  μV    10000              3000       0.11        n/a   LTG   mastoid   lateral_temporal_surface good   n/a
```

> LT02  ECOG  μV    10000              3000       0.11        n/a   LTG   mastod   lateral_temporal_surface bad    brokenn

Reviewer: ```suggestion
LT02  ECOG  μV    10000              3000       0.11        n/a   LTG   mastod    lateral_temporal_surface bad    broken
```

H01   SEEG  μV    10000              3000       0.11        n/a   HST   mastoid   hippocampal_depthh       bad    line noisee
ECG1  ECG   μV    10000              n/aa       0.11        60    4     ECG2      ecg_channell             good   na
TR1   TRIG  n/a   1000               n/a        n/a         n/a   5     n/a       ana_trigger              good   n/a
```
Restricted keyword list for field type in alphabetic order (shared with the MEG

> and EEG modality; however, only iEEG specific types are listed here):

Reviewer: It seems like we're listing more than just the iEEG specific types, no? Perhaps we should put them in bold? Or remove the non-iEEG specific types (like `EEG`)?

| PUPIL    | Eye Tracker pupil diameter                                        |
| MISC     | Miscellaneous                                                     |
| SYSCLOCK | System time showing elapsed time since trial started              |
| ADC      | Analog to Digital input                                           |
| DAC      | Digital to Analog output                                          |
| REF      | Reference channel                                                 |
| OTHER    | Any other type of channel                                         |

The free text field for the channel description can for example be specified as
intracranial, stimulus, response, vertical EOG, horizontal EOG, skin
conductance, eyetracker, etc.

## Electrode description table (`*[_space-<label>]_electrodes.tsv`)

Template:

```Text
sub-<label>/
    [ses-<label>]/
      ieeg/
         sub-<label>[_ses-<label>][_space-<label>]_electrodes.tsv
```

File that gives the location, size and other properties of iEEG electrodes. Note
that coordinates are expected in cartesian coordinates according to the
`iEEGCoordinateSystem` and `iEEGCoordinateSystemUnits` fields in

> `*_coordsystem.json`. **If an `*_electrodes.tsv` file is specified, a

Reviewer: ```suggestion
`*_coordsystem.json`.

**If an `*_electrodes.tsv` file is specified, a
```

> `*_coordsystem.json` file MUST be specified as well**. The order of the required

Reviewer: ```suggestion
`*_coordsystem.json` file MUST be specified as well**.

The order of the required
```

columns in the `*_electrodes.tsv` file MUST be as listed below.

> MUST be present:

Reviewer: ```suggestion
**MUST be present**

```

| :--------- | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| name       | REQUIRED. Name of the electrode contact point.                                                                                                                               |
| x          | REQUIRED. X position. The positions of the center of each electrode in xyz space. Units are in millimeters or pixels and are specified in \_\*space-<label>\_electrode.json. |
| y          | REQUIRED. Y position.                                                                                                                                                        |
| z          | REQUIRED. Z position. If electrodes are in 2D space this should be a column of n/a values.                                                                                   |
| size       | REQUIRED. Surface area of the electrode, in mm^2.                                                                                                                            |

> SHOULD be present:

Reviewer: ```suggestion
**SHOULD be present**
```

| Field name   | Definition                                                                                                                                                           |
| :----------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| material     | OPTIONAL. Material of the electrodes.                                                                                                                                |
| manufacturer | OPTIONAL. Recommended field to specify the manufacturer for each electrode. Can be used if electrodes were manufactured by more than one company.                    |
| group        | OPTIONAL. Optional field to specify the group that the electrode is a part of. Note that any group specified here should match a group specified in `_channels.tsv`. |
| hemisphere   | OPTIONAL. Optional field to specify the hemisphere in which the electrode is placed, one of [‘L’ or ‘R’] (use capital).                                              |

> MAY be present:

Reviewer: ```suggestion
**MAY be present**
```

| iEEGCoordinateProcessingReference   | RECOMMENDED. A reference to a paper that defines in more detail the method used to project or localize the electrodes.                                                                                                                                                                                                                                                                                                                                                                                                                     |

### Recommended 3D coordinate systems

It is preferred that electrodes are localized in a 3D coordinate system (with
respect to a pre- and/or post-operative anatomical MRI or CT scans or in a

> standard space as specified in BIDS Appendix VIII: preferred names of Coordinate

Reviewer: ```suggestion
standard space as specified in [BIDS Appendix VIII: preferred names of Coordinate
```

> systems, such as ACPC).

Reviewer: ```suggestion
systems](../99-appendices/08-coordinate-systems), such as ACPC).
```

### Allowed 2D coordinate systems

If electrodes are localized in 2D space (only x and y are specified and z is

> n/a), then the positions in this file must correspond to the locations expressed

Reviewer: ```suggestion
`n/a`), then the positions in this file must correspond to the locations expressed
```

in pixels on the photo/drawing/rendering of the electrodes on the brain. In this
case, coordinates must be (row,column) pairs, with (0,0) corresponding to the
upper left pixel and (N,0) corresponding to the lower left pixel.

> ### Multiple coordinate systems

Reviewer: ```suggestion

### Multiple coordinate systems
```

If electrode positions are known in multiple coordinate systems (e.g., MRI, CT
and MNI), these spaces can be distinguished by the optional `[_space-<label>]`
and `[_proc-<label>]` fields.

The optional space label (`[_space-<label>]_electrodes.tsv`) indicates the way

> in which electrode positions are interpreted, see [TODO: link] BEP003 - Common

Reviewer: ```suggestion
in which electrode positions are interpreted, see [BEP003 - Common
```

> Derivatives. Examples include:

Reviewer: ```suggestion
Derivatives](../06-extensions). Examples include:
```

-   `_space-orig` (electrodes are in the space originally extracted from the
    image, such as a T1 weighted MRI, CT, XRay or 2D operative photo).
-   `_space-MNI152Lin` (electrodes are coregistred and scaled to a specific MNI
    template)
-   `_space-Talairach` (electrodes are coregistred and scaled to Talairach
    space)

Example:

```json
{
    "IntendedFor": "/sub-01/ses-01/anat/sub-01_T1w.nii.gz",
    "iEEGCoordinateSystem": "ACPC",
    "iEEGCoordinateUnits": "mm",
    "iEEGCoordinateSystemDescription": "Coordinate system with the origin at anterior commissure (AC), negative y-axis going through the posterior commissure (PC), z-axis going to a mid-hemisperic point which lies superior to the AC-PC line, x-axis going to the right",
    "iEEGCoordinateProcessingDescription": "surface_projection",
    "iEEGCoordinateProcessingReference": "Hermes et al., 2010 JNeuroMeth"
}
```

## Photos of the electrode positions (`*_photo.jpg`)

Template:

```Text
sub-<label>/
    [ses-<label>]/
      ieeg/
         sub-<label>[_ses-<label>][_acq-<label]_photo.json
```

These can include photos of the electrodes on the brain surface, photos of

> anatomical features or landmarks (such as sulcal structure), fiducials. Photos

Reviewer: ```suggestion
anatomical features or landmarks (such as sulcal structure), and fiducials. Photos
```

can also include an X-ray picture, a flatbed scan of a schematic drawing made
during surgery, or screenshots of a brain rendering with electrode positions.
The photos may need to be cropped and/or blurred to conceal identifying features
or entirely omitted prior to sharing, depending on obtained consent.

If there are photos of the electrodes, the acquisition field should be specified
with:

-   `*_photo.jpg` in case of an operative photo
-   `*_acq-xray#_photo.jpg` in case of an x-ray picture
-   `*_acq-drawing#_photo.jpg` in case of a drawing or sketch of electrode
    placements
-   `*_acq-render#_photo.jpg` in case of a rendering

The session label may be used to specify when the photo was taken.

> Example of the operative photo of ECoG electrodes (here an annotated example in

Reviewer: ```suggestion
Example of the operative photo of ECoG electrodes (here is an annotated example in
```

which electrodes and vasculature are marked, taken from Hermes et al.,
JNeuroMeth 2010).

```
    sub-0001_ses-01_acq-photo1_photo.jpg
    sub-0001_ses-01_acq-photo2_photo.jpg
```

[TODO: brain picture]

Below is an example of a volume rendering of the cortical surface with a
superimposed subdural electrode implantation. This map is often provided by the
EEG technician and provided to the epileptologists (e.g., see Burneo JG et al.
2014 http://dx.doi.org/10.1016/j.clineuro.2014.03.020).

```
    sub-0002_ses-001_acq-render_photo.jpg (for volume rendering)
```

[TODO: brain picture 2]

> ## Electrical stimulation

Author: I think this addresses Q1 of https://github.com/bids-standard/bids-specification/issues/61

Author: @dorahermes and @choldgraf are you ok with this, or is more needed on electrical stimulation?

In case of electrical stimulation of brain tissue by passing current through the
iEEG electrodes, and the electrical stimulation has an event structure (on-off,
onset, duration), the `_events.tsv` file can contain the electrical stimulation
parameters in addition to other events. Note that these can be intermixed with
other task events. Electrical stimulation parameters can be described in columns

> called electrical*stimulation*<label>, with labels chosen by the researcher and

Reviewer: ```suggestion
called `electrical*stimulation*<label>`, with labels chosen by the researcher and
```
